osp/dpdk: fix vfio-pci config step (#43560)
openshift/openshift-docs: modules/installation-osp-dpdk-binding-vfio-pci.adoc

$ openstack network show <VFIO_network_name> -f value -c id
----

> . Create a Butane config file, `99-vhostuser-bind.bu`, binding the PCI device to the VFIO driver.

Reviewer: ```suggestion
. Create a Butane config file called `99-vhostuser-bind.bu` that binds the PCI device to the VFIO driver. As an example:
```

+
[NOTE]
====

> See "Creating machine configs with Butane" for information about Butane.

Reviewer: Link for this?

Author: I looked for other references of Butane in this repo and none of them have a link for reference. Please suggest if you want me to change this content.

Reviewer: It's nice when possible, but we cannot link from one module to another due to publishing toolchain limitations. :\ 

I couldn't tell from context alone if this was in the same docs set or not. Since it is, possible options are:

* Create an "Additional resources" section after this module in any assemblies that contain it. A link out can live there. 
* Add more context to the note to help with search. Ex: "For more information about Butane, see 'Creating machine configs with Butane' in the 'Customizing nodes' section of the documentation."  
* Leave it as is.

Author: I went with option 2, let me know if it's good.

> For more information about Butane, see 'Creating machine configs with Butane' in the 'Customizing nodes' section of the documentation.

Reviewer: ```suggestion
For more information about Butane, see "Creating machine configs with Butane" in the "Customizing nodes" section of the documentation.
```
Changing quote style. Single -> double outside of nested quotes context.

Thanks!

                        return 1
                    fi
        
                    if [ ! -f "$syspath/driver_override" ]; then
                        echo "File $syspath/driver_override not found"
                        return 1
                    fi
                    if ! echo "vfio-pci">"$syspath/driver_override"; then
                        return 1
                    fi
        
                    if [ ! -f "/sys/bus/pci/drivers/vfio-pci/bind" ]; then
                        echo "File /sys/bus/pci/drivers/vfio-pci/bind not found"
                        return 1
                    fi
                    if ! echo "$dev">"/sys/bus/pci/drivers/vfio-pci/bind"; then
                      return 1
                    fi
                    return 0
                fi
            done
            return 1
        }
        
        for nwid in "$@"; do
            parseNetwork $nwid
        done
    mode: 0744
    path: /usr/local/bin/vhostuser
  - contents:
      inline: |
        ARG="be22563c-041e-44a0-9cbd-aa391b439a39,ec200105-fb85-4181-a6af-35816da6baf7" <2> 
    mode: 0644
    path: /etc/vhostuser-bind.conf
----

> <1> Applies the new kernel argument only to worker nodes.

Reviewer: ```suggestion
<1> This label applies the new kernel argument to worker nodes only.
```

Or something like that.

<2> Replace this value with a comma-separated list of VFIO network UUIDs. 
====
+
On boot for machines that are part of this set, the MAC addresses of ports are translated into PCI bus IDs. The `vfio-pci` module is bound to any port that is assocated with a network that is identified by the {rh-openstack} network ID.

> . Use Butane to generate a `MachineConfig` object file, `99-vhostuser-bind.yaml`, containing the configuration to be delivered to the worker nodes:

Reviewer: ```suggestion
. From a command line, use Butane to generate a `MachineConfig` object file, `99-vhostuser-bind.yaml`, that contains a configuration to deliver to worker nodes:
```

+
[source,terminal]
----
$ butane 99-vhostuser-bind.bu -o 99-vhostuser-bind.yaml
----

. Apply the `MachineConfig` object to the worker nodes:
+
[source,terminal]
----
$ oc apply -f 99-vhostuser-bind.yaml
----

> . Verify that the `MachineConfig` object was added.

Reviewer: ```suggestion
. Verify that the `MachineConfig` object was added:
```
